### PR TITLE
Add commands to reset tutorials to not taken place 

### LINKED ISSUE #95: Allow users to set tutorials as not taken place
Users could potentially typo a tutorial number and mark/unmark a student for a tutorial that has not taken place yet. Currently, there is no option to revert a tutorial back to not taken place.

### PR DESCRIPTION
**Pull request by sayglenn**
Tutors might need to reset tutorials to the state of not taken place in case of typos. Added a ResetCommand to do so.

Resolves #95.

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/ResetCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        Model typicalModel = new ModelManager(getMarkedAddressBook(TUTORIAL_TO_TEST), new UserPrefs());
+        assertCommandSuccess(resetCommand, typicalModel, expectedMessage, expectedModel);
+    }
+
+    /**
+     * Reset all persons in the list.
+     */
+    @Test
+    public void execute_shouldResetAll_success() {
+        Tutorial tutorialToBeAdded = new Tutorial(TUTORIAL_TO_TEST);
```
Comment (Domokunx) [inline src/test/java/seedu/address/logic/commands/ResetCommandTest.java:103]: tutorialToBeAdded seems like a misleading name?

File: src/test/java/seedu/address/logic/commands/ResetCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            Person expectedEditedPerson = new Person(
+                    person.getName(),
+                    person.getStudentId(),
+                    person.getPhone(),
+                    person.getEmail(),
+                    person.getTags(),
+                    newTutorials
+            );
+
+            assertEquals(expectedEditedPerson, person);
```
Comment (Domokunx) [inline src/test/java/seedu/address/logic/commands/ResetCommandTest.java:126]: Might be easier to use assertCommandSuccess() here?

**Review by Domokunx — COMMENTED** [review]
Nothing seems wrong implementation wise!
Testcases are also written with a lot of coverage and breadth.
Just some cosmetic nits.

File: src/test/java/seedu/address/logic/commands/ResetCommandTest.java (hunk 1 above)
Comment (sayglenn) [reply src/test/java/seedu/address/logic/commands/ResetCommandTest.java]: Resolved, amended to tutorialToBeReset

File: src/test/java/seedu/address/logic/commands/ResetCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+import seedu.address.model.AddressBook;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.AttendanceStatus;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Tutorial;
+
+public class ResetCommandTest {
+    private static final String TUTORIAL_TO_TEST = "1";
```
[the remark this answers is not in the document]
Comment (sayglenn) [reply src/test/java/seedu/address/logic/commands/ResetCommandTest.java]: Good catch! Amended as such.

File: src/main/java/seedu/address/model/util/SampleDataUtil.java [hunk 4]
```diff
@@ -81,8 +81,6 @@ public static Map<Tutorial, AttendanceStatus> getTutorialMap(String ... strings)
     }
```
[the remark this answers is not in the document]
Comment (sayglenn) [reply src/main/java/seedu/address/model/util/SampleDataUtil.java:88]: Added JavaDocs!

File: src/main/java/seedu/address/logic/commands/ResetCommand.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+     * @param index The index of the person to be marked.
+     * @param tutorial The tutorial to set as not taken place.
+     */
+    public ResetCommand(Index index, Tutorial tutorial) {
+        this.index = index;
+        this.tutorial = tutorial;
+    }
+
+    /**
+     * Takes a person and tutorial and resets the specified tutorial and returns that person with the new tutorials.
```
Comment (Domokunx) [inline src/main/java/seedu/address/logic/commands/ResetCommand.java:43]: Empty line between method header and params

**Review by Domokunx — APPROVED** [review]
LGTM!

**Pull request merged**